cosmos/operators/virtualenv.py

```diff
@@ -76,14 +76,16 @@ def __init__(
         py_requirements: list[str] | None = None,
         pip_install_options: list[str] | None = None,
         py_system_site_packages: bool = False,
-        virtualenv_dir: Path | None = None,
+        virtualenv_dir: str | None = None,
```

Comment:
@pankajastro this could be seen as a breaking change - what motivated it?

Reply:
I pushed that change @tatiana in https://github.com/astronomer/astronomer-cosmos/pull/1721/commits/1f1947c8c2fe3b3eb52a9c23aa04ae4c8dbd7572. We had a discussion regarding this in the other PR at https://github.com/astronomer/astronomer-cosmos/pull/1718#discussion_r2065555001.

Apparently Airflow 3 is evaluating & validating that the `Path` exists as part of its rending as mentioned by Pankaj in the other PR's description.

Reply:
@pankajkoti I just read the thread - would it be possible for this to be a Path and for us to convert to a string if Airflow 3 only, as part of the operator implementation?

Reply:
we need to try out though, when does the rendering evaluate and if we have control over it(meaning if the rendering happens even before the operator initializes/executes and if yes, if there's a method to override the type for it).

Reply:
will try it out later today to dig into this.